spec/requests/citizens/consents_spec.rb

```diff
@@ -1,11 +1,42 @@
 require 'rails_helper'
 
-RSpec.describe 'citizen consents request test', type: :request do
-  describe 'GET /citizens/consent' do
-    before { get citizens_consent_path }
+RSpec.describe Applicants::OpenBankingConsentForm, type: :request do
+  describe 'citizen consent request test' do
+    describe 'GET /citizens/consent' do
+      before { get citizens_consent_path }
 
-    it 'returns http success' do
-      expect(response).to have_http_status(:ok)
+      it 'returns http success' do
+        expect(response).to have_http_status(:ok)
+        expect(unescaped_response_body).to include('I agree for you to check')
+      end
+    end
+  end
+
+  describe 'POST /citizens/additional_accounts', type: :request do
+    let(:applicant) { create :applicant }
+    let(:legal_aid_application) { create :legal_aid_application, applicant: applicant }
+    before do
+      sign_in applicant
+      post citizens_consent_path, params: params
+    end
+
+    context 'when consent is granted' do
+      let(:params) { { legal_aid_application: { open_banking_consent: 'YES' } } }
+
+      it 'redirects to new action' do
+        expect(response).to redirect_to(applicant_true_layer_omniauth_authorize_path)
+        expect(unescaped_response_body).to include('true')
+      end
```

Comment:
I think there should also be a spec to check that the decisions have been persisted. Something like:
```ruby
it 'records decision on the legal aid application' do
  legal_aid_application.reload
  expect(legal_aid_application.open_banking_consent).to eq(true)
  expect(legal_aid_application.open_banking_consent_choice_at.to be_between(2.seconds.ago, 1.second.from_now)
end
```

Reply:
👍